perun-core/src/main/java/cz/metacentrum/perun/core/implApi/MembersManagerImplApi.java

```diff
@@ -215,6 +236,15 @@
 	 */
 	void removeSponsor(PerunSession sess, Member sponsoredMember, User sponsor) throws InternalErrorException;
 
+	/**
+	 * Delete all existing sponsorships for defined member. This method will delete them, not just marked.
+	 *
+	 * @param session perun session
+	 * @param sponsoredMember member which is sponsored
+	 * @throws InternalErrorException
+	 */
+	void deleteAllSponsors(PerunSession session, Member sponsoredMember) throws InternalErrorException;
```

Comment:
This method should not exist on the database level, vis. my comment for its implementation.

Reply:
The only purpose of this method is to delete all existing connections of member to his sponsors. It is a problem of methods from business logic to take care of needed audit messages.